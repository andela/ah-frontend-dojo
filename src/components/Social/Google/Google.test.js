// react libraries
import React from 'react';
// third-party libraries
import { shallow } from 'enzyme';
// components
import GoogleButtons from './index';
// Define a shared shallow renderer function
const googleBtn = (props = {}) => {
  const google = shallow(<GoogleButtons {...props} />);
  return google;
};
// Define the tests for the Facebook button
describe('Google Login', () => {
  // Pass the value of the shallow renderer before each test

  let google;
  beforeEach(() => {
    google = googleBtn();
  });

  // Test to see if it matches the snapshot

  it('should match the snapshot', () => {
    const wrapper = shallow(<GoogleButtons />);
    expect(wrapper).toMatchSnapshot();
  });

  // Test to see if it works as expected
  it('should work as expected', () => {
    const response = {
      accessToken: 'auth_token'
    };
    google.instance().responseGoogle(response);
  });

  // Test to see if it renders a facebook button
  it('should render a google button', () => {
    const wrapper = google.find('.googlebutton');
    expect(wrapper.length).toBe(1);
  });
});
