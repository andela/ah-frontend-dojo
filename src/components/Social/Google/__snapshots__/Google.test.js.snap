// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Google Login should match the snapshot 1`] = `
<div
  className="googlebutton"
  id="googlebutton"
>
  <e
    accessType="online"
    buttonText={false}
    clientId="39137008692-dmdrj00qulao069le80rk6rores32vdh.apps.googleusercontent.com"
    cookiePolicy="single_host_origin"
    disabledStyle={
      Object {
        "opacity": 0.6,
      }
    }
    fetchBasicProfile={true}
    icon={
      <img
        alt="Google"
        src="https://icons8.com/icon/17949/google"
      />
    }
    isSignedIn={false}
    jsSrc="https://apis.google.com/js/api.js"
    onFailure={[Function]}
    onRequest={[Function]}
    onSuccess={[Function]}
    prompt=""
    scope="profile email"
    tag="button"
    theme="light"
    type="button"
    uxMode="popup"
  />
</div>
`;
